src/blocks/custom.rs

```diff
@@ -141,7 +148,16 @@ async fn update_bar(
 
 pub async fn run(config: toml::Value, mut api: CommonApi) -> Result<()> {
     let config = CustomConfig::deserialize(config).config_error()?;
-    let mut widget = api.new_widget();
+    let mut format_config = FormatConfig { full: config.format.full, short: config.format.short };
+    let format = if config.json {
+        if format_config.short.is_none() {
+            format_config.short = Some(" $icon $short_text ".parse()?);
+        };
+        format_config.with_default(" $icon $text ")?
```

Comment:
I think setting the default to `{ $icon|} $text {$short_text |}` would be sufficient?

Reply:
I don't think I'm able to maintain the same behavior as on master with any configuration but I like your suggestion much more than my conditions...

I'm using the following config for testing:
```toml
[[block]]
block = "custom"
json = true
command = "echo '{\"icon\": \"cpu\", \"text\": \"!\", \"short_text\": \"?\", \"state\": \"Critical\"}'"

[[block]]
block = "custom"
json = true
command = "echo '{\"icon\": \"cpu\", \"text\": \"#\", \"state\": \"Critical\"}'"

[[block]]
block = "custom"
command = "echo '--'"
```

On master this gives me:

```json
[
    {"full_text":"  "},
    {"full_text":"! ","short_text":"<span/>"},
    {"full_text":"<span/>","short_text":"? "},
    {"full_text":"  ","color":"#002B36FF"},
    {"full_text":"# "}
]
```

Testing it with `with_default("{ $icon|} $text {$short_text |}")?` and `[if let Some(t) = input.short_text] "short_text" => Value::text(t)` gives me:

```json
  [
    {"full_text":"  ! "},
    {"full_text":"? "},
    {"full_text":"  # "},
    {"full_text":"  "}
  ]
```

I think the best solution here would be to use `"{ $icon|} $text"` as default but in that case, we need to set default short formatting option when json output `short_text` field is present. I've tried to do that but I'm struggling with mutating `short` FormatConfig field inside `update_bar` without adding a `Copy` trait to FormatConfig :thinking:  :thinking:

Reply:
Yeah, you're right. Up until now there weren't a need to set default short text. But I think this should be good:

```rust
let mut widget = api.new_widget().with_format(
    config
        .format
        .with_default("{ $icon|} $text ")?
        .with_short_default("$short_text|")?
);
```

Reply:
:thinking: :thinking: :thinking: 

I'm unsure about adding `with_short_default`. `with_default` is FormatConfig method and it returns `Result<Format>`. If I were to add  `with_short_default` and keep it on the same level, we'd have to change return value to `Self` but then it's also not pretty... I've also thought about changing the signature from `with_default(self, default_full: &str) -> Result<Format>` to `with_default(self, default_full: &str, default_short: Option<&str>) -> Result<Format>` and updating ALL but custom block to pass  'None` but decided against because it would make this PR much larger.

I think I've got it:

```toml
[[block]]
block = "custom"
json = true
command = "echo '{\"icon\": \"cpu\", \"text\": \"!\", \"short_text\": \"?\", \"state\": \"Critical\"}'"

[[block]]
block = "custom"
json = true
command = "echo '{\"icon\": \"cpu\", \"text\": \"#\", \"state\": \"Critical\"}'"

[[block]]
block = "custom"
command = "echo 'abc'"
```
Gives me:

```json
    { "full_text": "  ! ", "short_text": "<span/>", "name": "0"},
    { "full_text": "<span/>", "short_text": " ? ", "name": "0"},
    { "full_text": "  # ", "name": "1" },
    { "full_text": " abc ", "name": "2" }
```

It's not pretty but doesn't require that many changes on this PR.
```rust
    let mut format_config = FormatConfig { full: config.format.full, short: config.format.short };
    if format_config.short.is_none() {
        format_config.short = Some(" $short_text |".parse()?);
    };
    let mut widget = api.new_widget().with_format(
        format_config.with_default("{ $icon|} $text ")?
    );
```

Reply:
Maybe add `with_defaults(full: &str, short: &str)` (similar to `.set_texts()`)?

Reply:
Done, added in https://github.com/greshake/i3status-rust/pull/1618/commits/249bc4030017094767b3c39a619d179b78b94ac1